File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.py
def solution(lottos, win_nums):
    # 로또 순위 해시값으로 설정(검색, 수정 빠르게 하기 위해)
    rank={6:1,5:2,4:3,3:4,2:5}
    # 0개수, 0이 아니지만 당첨 번호와 불일치 개수, 로또 구매 번호 길이 변수 설정
    z,o,l=0,0,len(lottos)
    # 로또 번호 리스트 돌면서
    for n in lottos:
        # 번호가 0보다 크고 로또 번호와 불일치할 때
        if n>0 and n not in win_nums:
            o+=1
        # 번호가 0일 때
        elif n==0:
            z+=1
    # rank 해시에 일치된 개수 보내서 순위 도출 
    r=[rank.get(l-o,6),rank.get(l-z-o,6)]
    return r

File: 프로그래머스/1/77484. 로또의 최고 순위와 최저 순위/로또의 최고 순위와 최저 순위.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

// lottos_len은 배열 lottos의 길이입니다.
// win_nums_len은 배열 win_nums의 길이입니다.
int* solution(int lottos[], size_t lottos_len, int win_nums[], size_t win_nums_len) {
    // return 값은 malloc 등 동적 할당을 사용해주세요. 할당 길이는 상황에 맞게 변경해주세요.
    int* r=(int*)malloc(sizeof(int)*2);
    memset(r,0,sizeof(int)*2);
    int zeros=0,matches=0,rank[7]={6,6,5,4,3,2,1};
    for(int i=0;i<lottos_len;i++) {
        if(lottos[i]==0)
            zeros+=1;
        for(int j=0;j<win_nums_len;j++) {
            if(lottos[i]==win_nums[j]) {
                matches+=1;
                break;
            }
        }
    }
    r[0]=rank[zeros+matches];
    r[1]=rank[matches];
    return r;
}